extensions/vscode/src/activation/activate.ts

```diff
@@ -52,14 +53,23 @@ export async function activateExtension(context: vscode.ExtensionContext) {
   }
 
   const api = new VsCodeContinueApi(vscodeExtension);
-  const showOnboarding = !isGraniteOnboardingComplete(context);
-  vscode.commands.executeCommand(
+  const graniteOnboardingComplete = isGraniteOnboardingComplete(context);
+  await vscode.commands.executeCommand(
     "setContext",
     "granite.initialized",
-    !showOnboarding,
+    graniteOnboardingComplete,
   );
-  if (showOnboarding) {
-    await vscode.commands.executeCommand("granite.setup");
+
+  const showOnboardingAtStartup = !context.globalState.get<boolean>(GRANITE_INITIAL_ACTIVATION_COMPLETED_KEY, false);
+  if (graniteOnboardingComplete) {
+    if (!showOnboardingAtStartup) {
+      await context.globalState.update(GRANITE_INITIAL_ACTIVATION_COMPLETED_KEY, true);
+    }
+  } else {
+    if (showOnboardingAtStartup) {
+      await context.globalState.update(GRANITE_INITIAL_ACTIVATION_COMPLETED_KEY, true);
+      await vscode.commands.executeCommand("granite.setup");
+    }
```

Comment:
What about:

```ts
const initialActivationCompleted = context.globalState.get<boolean>(GRANITE_INITIAL_ACTIVATION_COMPLETED_KEY, false);
if (!initialActivationCompleted) {
  if (!graniteOnboardingComplete) {
          await vscode.commands.executeCommand("granite.setup");;
  }
  await context.globalState.update(GRANITE_INITIAL_ACTIVATION_COMPLETED_KEY, true);
}
```
The fact that `showOnboardingAtStartup` is reversed from the key makes it hard for me to follow.

Reply:
Sorry not sure I how got to that mess, your solution is 100% better